src/transformers/pipelines/token_classification.py

```diff
@@ -234,56 +234,68 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
+                    label = self.model.config.id2label[max_label_idx]
+                elif strategy == "average":
+                    max_label_idx = np.mean(score, axis=0).argmax()
+                    label = self.model.config.id2label[max_label_idx]
+                else:
+                    raise ValueError(f"Invalid value {strategy} for option `subword_label_re_alignment`")
 
-            if strategy == "first":
-                # get label of first sub-word
-                label_idx = score[0].argmax()
-                label = self.model.config.id2label[label_idx]
-            elif strategy == "max":
-                max_label_idx = np.unravel_index(np.argmax(score, axis=None), score.shape)[1]
-                label = self.model.config.id2label[max_label_idx]
-            elif strategy == "average":
-                max_label_idx = np.mean(score, axis=0).argmax()
-                label = self.model.config.id2label[max_label_idx]
+                for idx, sub in enumerate(sub_words):
+                    sub["entity"] = label
+                    sub["score"] = score[idx][max_label_idx].item()
 
-            for idx, sub in enumerate(sub_words):
-                sub["entity"] = label
-                sub["score"] = score[idx][max_label_idx].item()
+            if self.ignore_subwords:
+                sub_words[0]["word"] += "".join([sub["word"].split("##")[1] for sub in sub_words[1:]])
+                return [sub_words[0]]
 
             return sub_words
 
-        word_group_disagg = []
-        entities_with_label = []
-
-        for entity in reversed(entities):
-            is_subword = entity["is_subword"]
-            word_group_disagg += [entity]
-            if not is_subword:
-                begin_sub = word_group_disagg.pop(-1)
-                if len(word_group_disagg) and word_group_disagg[0]["is_subword"]:
-                    word_group_disagg.reverse()
-                    merged_entity = sub_words_label(sub_words=word_group_disagg)
-                    entities_with_label.extend(merged_entity)
-
-                label_idx = begin_sub["score"].argmax()
-                label = self.model.config.id2label[label_idx]
-                begin_sub["entity"] = label
-                begin_sub["score"] = begin_sub["score"][label_idx].item()
+        subword_indices = np.where([entity["is_subword"] for entity in entities])[0]
+        if subword_indices.size == 0:
+            adjacent_subwords = []
+        else:
+            # find non-consecutive indices to identify separate clusters of subwords
+            cluster_edges = np.where(np.diff(subword_indices) != 1)[0]
+            # Sets of adjacent subwords indices, e.g.
+            # ['Sir', 'Test', '##y', 'M', '##c', '##T', '##est', 'is', 'test', '##iful']
+            # --> [[2],[4,5,6],[9]]
+            adjacent_subwords = np.split(subword_indices, cluster_edges + 1)  # shift edge by 1
+
```

Comment:
love your solution to find adjacent subwords 👍👍. I knew that we can probably do it with numpy but didn't figure it out